src/main/java/org/spdx/sbom/gradle/SpdxSbomTask.java

```diff
@@ -84,6 +96,36 @@ public abstract class SpdxSbomTask extends DefaultTask {
 
   @TaskAction
   public void generateSbom() throws Exception {
+    Set<ComponentIdentifier> componentIds = new HashSet<>();
+    for (var rootComponent : getRootComponents().get()) {
+      componentIds.addAll(
+          gatherSelectedDependencies(
+              rootComponent,
+              new HashSet<>(),
+              new HashSet<>()
+          )
+      );
+    }
+
+    Map<ComponentArtifactIdentifier, File> resolvedArtifactsById = new HashMap<>();
+
+    @SuppressWarnings("unchecked")
+    List<ResolvedArtifactResult> resolvedArtifactResults =
+        getDependencyHandler()
+            .createArtifactResolutionQuery()
+            .forComponents(componentIds)
+            .withArtifacts(MavenModule.class, MavenPomArtifact.class)
+            .execute()
+            .getResolvedComponents()
+            .stream()
+            .flatMap(componentArtifactsResult -> componentArtifactsResult.getArtifacts(MavenPomArtifact.class).stream())
+            .filter(artifactResult -> artifactResult instanceof ResolvedArtifactResult)
+            .map(artifactResult -> (ResolvedArtifactResult) artifactResult)
+            .peek(resolvedArtifactResult -> resolvedArtifactsById.put(resolvedArtifactResult.getId(), resolvedArtifactResult.getFile()))
+            .collect(Collectors.toList());
+
+    PomResolver pomResolver = PomResolver.newPomResolver(getDependencyHandler(), getConfigurations(), getLogger());
```

Comment:
The tests and functionalTests both pass with this, and my project looks OK with it. I think it should be fine, but I'm not 100% sure.